### ISSUE TITLE
apply_to_vector to lists of vectors raises ValueError("Vector size unsupported")

### ISSUE THREAD

**Issue by antoniotejada** [adamlwgriffiths/Pyrr#106]
The documentation states that apply_to_vector can be used to transform a list of vectors (instead of calling apply_to_vector on each one in a Python loop)
```python
"""
    :param numpy.array vec: The vector to modify.
        Can be a list of vectors.
"""
```
https://github.com/adamlwgriffiths/Pyrr/blob/34802ba0393a6e7752cf55fadecd0d7824042dc0/pyrr/matrix44.py#L196

Unfortunately the vec.size check in both matrix33 and matrix44 implementations above prevents that and raises a ValueError("Vector size unsupported")

calling apply_to_vector in a loop is 12x slower than using numpy natively
´´´
Numpy
1000000 vectors in 0.53 seconds, 1893938.98 vectors/second
Pyrr no conversion
1000000 vectors in 6.20 seconds, 161238.31 vectors/second
´´´

**Comment by adamlwgriffiths** [comment]
Some changes were applied in #108, can you test if the latest changes fix this for you?

**Comment by antoniotejada** [comment]
Thanks @adamlwgriffiths , unfortunately it's going to take me some time to get back to this, although I assume that change would fix it.

If you are curious about performance I gathered [a bunch of comparisons](https://github.com/antoniotejada/epycc#performance) for my [epycc](https://github.com/antoniotejada/epycc) project.

**Comment by adamlwgriffiths** [comment]
No rush.
Project looks _very_ cool, thanks for sharing

**Comment by antoniotejada** [comment]
Thanks and likewise! I used pyrr heavily in some weird STL / editor / 3D graphics playground I'm writing, it made things a lot easier than straight numpy's learning curve.

The slowness of Python smoothing normals even with numpy-optimized algorithms is what made me create epycc. With numpy you are SOL if your algorithm doesn't fit it, and numba is cool but the little I've used it I've found it to be a lot of guesswork and trial and error to appease it (also the codegen sucks donkey balls as you can see in epycc's performance study).

Hopefully one day I'll get my lazy ass around to push my playground to github too.

![image](https://user-images.githubusercontent.com/6446344/135779523-4804ffdf-ee6e-431e-8332-42558fadd98b.png)
(STLs from mz4250)

**Comment by adamlwgriffiths** [comment]
With numpy you really need to vectorise your data so you can do mass transforms.
Depending on how the data already is laid out, it can be expensive, but it is the better way to do things en-masse.
You'd probably be better off doing it at either:
* run-time via shader,
* load time with caching (and potential re-export), like shader caches.
Most mesh formats are really not suitable to direct rendering.

Although I'm sure you already figured these out.
Looks very cool!

**Comment by antoniotejada** [comment]
Right, so STL is actually pretty friendly, it's just a vertexbuffer and a normalbuffer of independent tris, no indices so no shared vertices across faces. This means there's a lot of vertex duplication (~12x in my stats) and normals are per face, not per vertex, so it looks very faceted.

![image](https://user-images.githubusercontent.com/6446344/135907563-5a324c6f-284d-495f-8f99-54ac94148780.png)
![image](https://user-images.githubusercontent.com/6446344/135907581-70b594d2-ca7d-41b3-9437-44bba2b0c940.png)

I wrote several numpy vectorized algos to do normal sharing/smoothing across faces with angle thresholding, so only normals that are "close enough" are smoothened out and then shared across faces, which not only makes the model look better, but also reduces the vertex count since now you can share vertices (and normals) across most faces:

![image](https://user-images.githubusercontent.com/6446344/135907605-e86c1547-7368-44b1-bbf3-62a69541a6f0.png)
![image](https://user-images.githubusercontent.com/6446344/135907623-67f1816b-73b1-4145-952d-aa2d5ae6afbe.png)

The fastest numpy vectorized algo I wrote was an xyz vertex sort to put the repeated vertices together, and then do predicate masking to sum all the "close enough" normals depending on the angle between normals, and that way get the shared normal value. Still, that takes around 3-5s for that simple model of 200K vertices on my laptop.
